### PR TITLE
Dockerization of this application

### PR DESCRIPTION
**Pull request by Basalisk9271**
I added a Dockerfile and docker-compose.yml so that it was easy to spin this application up in its own docker container. I also changed the README to make sure that the Docker information was in there.

### CODE REVIEW

**Comment by Basalisk9271** [comment]
Most recent change is the creation of the Login Modal. Unfortunately, I have not gotten the Database working yet, but once I do, then we can start testing the User session and login stuff.

File: README.md [hunk 1]
```diff
@@ -93,3 +93,11 @@ yarn start
 ## Stopping Developement Server
 
 To stop the server, use <kbd>Ctrl</kbd> + <kbd>C</kbd>.
+
+## USing DOcker to containerize while working 
+
+To spin up a docker image of the application, use the docker-compose.yml file by running
+```
+docker-compose up --build --force-recreate
```
Comment (SnubbullTrouble) [inline README.md]: Running this command didn't work for me. Is docker-compose supposed to be a npm package? Do we have to install that separately?

File: src/app/_components/NavBar.tsx [hunk 2]
```diff
@@ -62,7 +62,7 @@ const Navbar = ({ user }: Props) => {
           variant="h6"
           sx={{ display: { xs: "none", sm: "block" } }}
         >
-          Who Wants this Dog?
+          Macon Pet Finder
```
Comment (SnubbullTrouble) [inline src/app/_components/NavBar.tsx]: This should be a separate branch/ticket.

File: src/app/_components/NavBar.tsx
Comment (SnubbullTrouble) [inline src/app/_components/NavBar.tsx:1]: The login modal should be on a different branch for tracking reasons.

File: src/app/_components/NavBar.tsx [hunk 3]
```diff
@@ -1,90 +1,84 @@
...
-            <Mail />
-          </Badge>
-          <Badge badgeContent={2} color="error">
-            <Notifications />
-          </Badge>
-          <Avatar
-            sx={{ width: 30, height: 30 }}
-            src={user.image}
-            onClick={() => setOpen(true)}
+        <Box sx={{ flexGrow: 1 }} />
```
Comment (SnubbullTrouble) [inline src/app/_components/NavBar.tsx]: Updating the search bar (which I think is what is happening here) should be a separate branch/ticket.

File: src/app/_components/NavBar.tsx [hunk 4]
```diff
@@ -1,90 +1,84 @@
...
+              "&::placeholder": { color: "lightgray" },
+            }}
           />
-        </Icons>
-        <UserBox onClick={() => setOpen(true)}>
-          <Avatar sx={{ width: 30, height: 30 }} src={user.image} />
-          <Typography component="span">{user.username}</Typography>
-        </UserBox>
-      </StyledToolbar>
+        </Box>
```
Comment (SnubbullTrouble) [inline src/app/_components/NavBar.tsx]: The modal stuff should be a separate branch/ticket.

File: src/app/_components/NavBar.tsx [hunk 5]
```diff
@@ -36,7 +36,7 @@ const Navbar: React.FC<Props> = ({ user }) => {
   return (
     <AppBar position="sticky">
       <Toolbar>
-        <img src={"../../public/logo.png"} width={50} height={50} alt="logo" />
+        <img src={"logo.png"} width={50} height={50} alt="logo" />
```
Comment (SnubbullTrouble) [inline src/app/_components/NavBar.tsx]: What was the issue with the icon? This should also be a separate branch/ticket.

**Comment by SnubbullTrouble** [comment]
@Basalisk9271 This should be done on its own branch.

> Most recent change is the creation of the Login Modal. Unfortunately, I have not gotten the Database working yet, but once I do, then we can start testing the User session and login stuff.

**Review by SnubbullTrouble — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Basalisk9271) [reply README.md]: Docker for my linux box did not come with it, so I had to apt install docker-compose on my server. I think Windows and Mac versions have a downloadable version. According to this site, the docker compose function should come with the Docker for Desktop installation: https://docs.docker.com/compose/install/

File: src/app/_components/NavBar.tsx (hunk 5 above)
Comment (Basalisk9271) [reply src/app/_components/NavBar.tsx]: For some reason, it was not loading in for me when I spun up the app. I fixed it by making the change and then it showed up for me.

File: README.md (hunk 1 above)
Comment (Basalisk9271) [reply README.md]: I also had to throw the env file as well as do the yarn install command before running the docker-compose command.

Comment (SnubbullTrouble) [reply README.md]: Should probably add that to your readme instructions.

**Pull request closed without merging**